KAFKA-3748: Add consumer-property to console tools consumer

bharatviswa504 opened the pull request.

Commit:
KAFKA-3748: Add consumer-property to console tools consumer

Commit:
KAFKA-3748: Add consumer-property to console tools consumer

reftel committed:
MINOR; Close parenthesis in PartitionInfo.toString

Trivial patch
is trivial.

Author: Magnus Reftel <magnus.reftel@skatteetaten.no>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1458 from magnusr/feature/paren

fpj committed:
KAFKA-3660: Log exception message in ControllerBrokerRequestBatch

Author: Flavio Junqueira <fpj@apache.org>

Reviewers: Grant Henke <granthenke@gmail.com>, Ismael Juma <ismael@juma.me.uk>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #1325 from fpj/KAFKA-3660

izeye committed:
MINOR: Add a newline to a Javadoc sample in KafkaConsumer

This PR simply adds a newline to a Javadoc sample in `KafkaConsumer` to flush the outputs.

Author: Johnny Lim <izeye@naver.com>

Reviewers: Ewen Cheslack-Postava <ewen@confluent.io>

Closes #1390 from izeye/patch-1

vahidhashemian committed:
KAFKA-3774: Make 'time' an optional argument of GetOffsetShell

Since the 'time' argument has a default value of -1, it makes sense to make it an optional argument.

Author: Vahid Hashemian <vahidhashemian@us.ibm.com>

Reviewers: Ewen Cheslack-Postava <ewen@confluent.io>

Closes #1457 from vahidhashemian/KAFKA-3774

edoardocomar commented on core/src/main/scala/kafka/tools/ConsoleConsumer.scala:
```diff
@@ -212,7 +214,11 @@ object ConsoleConsumer extends Logging {
       .describedAs("urls")
       .ofType(classOf[String])
-    val consumerConfigOpt = parser.accepts("consumer.config", "Consumer config properties file.")
+    val consumerPropertyOpt = parser.accepts("consumer-property", "A mechanism to pass user-defined properties in the form key=value to the consumer. ")
```
an extra space at the end

edoardocomar commented:
Thanks, I prefer this solution to #1409

mjsax committed:
HOTFIX: updated JavaDoc example for 0.9 tech-prev to 0.10

Author: Matthias J. Sax <matthias@confluent.io>

Reviewers: Sriharsha Chintalapani, Ismael Juma, Guozhang Wang

Closes #1450 from mjsax/hotfix

ijuma committed:
MINOR: Fix setting of ACLs and ZK shutdown in test harnesses

I found both issues while investigating the issue described in PR #1425.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Sriharsha Chintalapani <schintalapani@hortonworks.com>, Jun Rao <junrao@gmail.com>

Closes #1455 from ijuma/fix-integration-test-harness-and-zk-test-harness

gfodor committed:
KAFKA-3785; Fetcher spending unnecessary time during metrics recording

Author: Greg Fodor <gfodor@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1464 from gfodor/gfodor/kafka-3785

guozhangwang committed:
MINOR: Add user overridden test logging events

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1456 from guozhangwang/Kminor-test-logging

sasakitoa committed:
MINOR: set charset of Javadoc to UTF-8

Currently javadoc doesn't specify charset.
This pull reqeust will set this to UTF-8.

Author: Sasaki Toru <sasakitoa@nttdata.co.jp>

Reviewers: Ewen Cheslack-Postava <ewen@confluent.io>

Closes #1413 from sasakitoa/javadoc_garbled

trybak committed:
KAFKA-3784: TimeWindows#windowsFor calculation is incorrect

- Fixed the logic calculating the windows that are affected by a new …event in the case of hopping windows and a small overlap.
- Added a unit test that tests for the issue

Author: Tom Rybak <trybak@gmail.com>

Reviewers: Michael G. Noll, Matthias J. Sax, Guozhang Wang

Closes #1462 from trybak/bugfix/KAFKA-3784-TimeWindows#windowsFor-false-positives

edoardocomar committed:
KAFKA-3728; EndToEndAuthorizationTest offsets_topic misconfigured

Set OffsetsTopicReplicationFactorProp to 3 like MinInSyncReplicasProp  Else a consumer was able to consume via assign but not via subscribe, so the testProduceAndConsume is now duplicated to check both paths

Author: Edoardo Comar <ecomar@uk.ibm.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1425 from edoardocomar/KAFKA-3728

arunmahadevan committed:
KAFKA-3645; Fix ConsumerGroupCommand and ConsumerOffsetChecker to correctly read endpoint info from ZK

The host and port entries under /brokers/ids/<bid> gets filled only for PLAINTEXT security protocol. For other protocols the host is null and the actual endpoint is under "endpoints". This causes NPE when running the consumer group and offset checker scripts in a kerberized env. By always reading the host and port values from the "endpoint", a more meaningful exception would be thrown rather than a NPE.

Author: Arun Mahadevan <aiyer@hortonworks.com>

Reviewers: Sriharsha Chintalapani <schintalapani@hortonworks.com>, Ismael Juma <ismael@juma.me.uk>

Closes #1301 from arunmahadevan/cg_kerb_fix

Commit:
KAFKA-3768; Replace all pattern match on boolean value by if/else block.

Author: Satendra kumar <satendra@knoldus.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1445 from satendrakumar06/remove_boolean_pattern_match

granthenke committed:
KAFKA-3789; Upgrade Snappy to fix snappy decompression errors

Author: Grant Henke <granthenke@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1467 from granthenke/snappy-fix

xiaotao183 committed:
KAFKA-3787; Preserve the message timestamp in mirror maker

The timestamp of messages consumed by mirror maker is not preserved after sending to target cluster. The correct behavior is to keep create timestamp the same in both source and target clusters.

Author: Tao Xiao <xiaotao183@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com>, Ismael Juma <ismael@juma.me.uk>

Closes #1466 from xiaotao183/KAFKA-3787

ijuma committed:
MINOR: Fix producer leak in `PlaintextProducerSendTest`

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Sriharsha Chintalapani <harsha@hortonworks.com>, Guozhang Wang <wangguoz@gmail.com>

Closes #1471 from ijuma/fix-leaking-producers-in-plaintext-producer-send-test

ijuma committed:
MINOR: Move `ConsoleProducerTest` to the unit test directory

Included a couple of clean-ups: removed unused variable and the instantiated `KafkaProducer` is now closed.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Guozhang Wang <wangguoz@gmail.com>, Sriharsha Chintalapani <harsha@hortonworks.com>

Closes #1470 from ijuma/move-console-producer-test-to-unit-folder

rekhajoshm committed:
KAFKA-3771; Improving Kafka core code

- Used flatMap instead of map and flatten
- Use isEmpty, NonEmpty, isDefined as appropriate
- Used head, keys and keySet where appropriate
- Used contains, diff and find where appropriate
- Removed redundant val modifier for case class constructor
- toString has no parameters, no side effect hence without () consistent usage
- Removed unnecessary return , parentheses and semi colons.

Author: Joshi <rekhajoshm@gmail.com>
Author: Rekha Joshi <rekhajoshm@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1451 from rekhajoshm/KAFKA-3771

Commit:
KAFKA-3748: Add consumer-property to console tools consumer

Commit:
KAFKA-3748: Add consumer-property to console tools consumer
